internal/test/builder/builders.go

```diff
@@ -670,6 +715,63 @@ func (c *ControlPlaneTemplateBuilder) Build() *unstructured.Unstructured {
 	return obj
 }
 
+// TestControlPlaneTemplateBuilder holds the variables and objects needed to build a generic ControlPlane template.
+// +kubebuilder:object:generate=false
+type TestControlPlaneTemplateBuilder struct {
+	namespace                     string
+	name                          string
+	infrastructureMachineTemplate *unstructured.Unstructured
+	specFields                    map[string]interface{}
+}
+
+// TestControlPlaneTemplate creates a NewControlPlaneTemplate builder with the given name and namespace.
+func TestControlPlaneTemplate(namespace, name string) *TestControlPlaneTemplateBuilder {
```

Comment:
Can this be made deep-copyable? Like in #6016 

I'd be happy to do this in a follow-up if you don't have time.

Reply:
Thanks!